x-pack/plugins/security_solution/public/detection_engine/rule_monitoring/components/basic/indicators/execution_status_indicator/index.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiHealth } from '@elastic/eui';
+
+import type { RuleExecutionStatus } from '../../../../../../../common/detection_engine/rule_monitoring';
+import { getEmptyTagValue } from '../../../../../../common/components/empty_value';
+import { RuleStatusBadge } from '../../../../../../detections/components/rules/rule_execution_status';
+import {
+  getCapitalizedStatusText,
+  getStatusColor,
+} from '../../../../../../detections/components/rules/rule_execution_status/utils';
+
+interface ExecutionStatusIndicatorProps {
+  status?: RuleExecutionStatus | null | undefined;
+  showTooltip?: boolean;
+}
+
+const ExecutionStatusIndicatorComponent: React.FC<ExecutionStatusIndicatorProps> = ({
+  status,
+  showTooltip = false,
+}) => {
+  const statusText = getCapitalizedStatusText(status) ?? getEmptyTagValue();
```

Comment:
Discussed with @xcrzx. Not necessary in this case because the component is wrapped with React.memo, but it makes sense in general as a good practice. So, will do 👍